src/Middleware/HttpLogging/test/HttpLoggingMiddlewareTests.cs

```diff
@@ -1272,14 +1509,87 @@ public async Task HttpLoggingExtensionCanModifyRequestAndResponseSizeOnEndpoint(
         Assert.Contains(filteredLogs, w => w.Message.Equals("ResponseBody: testin"));
     }
 
+    [Fact]
+    public async Task InterceptorCanSeeAndOverrideExtensions()
+    {
+        var app = CreateApp(HttpLoggingFields.None, new FakeInterceptor(requestContext =>
+        {
+            Assert.Equal(HttpLoggingFields.All, requestContext.LoggingFields);
+            requestContext.Disable(HttpLoggingFields.RequestHeaders);
+        },
+        responseContext =>
+        {
+            Assert.Equal(HttpLoggingFields.All & ~HttpLoggingFields.RequestHeaders, responseContext.LoggingFields);
+            responseContext.Disable(HttpLoggingFields.ResponseHeaders);
+        }));
+        await app.StartAsync();
+
+        using var server = app.GetTestServer();
+        var client = server.CreateClient();
+        var request = new HttpRequestMessage(HttpMethod.Get, "/ext_responseandrequest");
+        request.Headers.TryAddWithoutValidation("Accept", "application/json");
+        var initialResponse = await client.SendAsync(request);
+
+        var filteredLogs = TestSink.Writes.Where(w => w.LoggerName.Contains("HttpLogging"));
+        Assert.Contains(filteredLogs, w => w.Message.Contains("Request"));
+        Assert.DoesNotContain(filteredLogs, w => w.Message.Contains("Accept"));
+        Assert.Contains(filteredLogs, w => w.Message.Contains("StatusCode: 200"));
+        Assert.DoesNotContain(filteredLogs, w => w.Message.Contains("Content-Type: text/plain"));
+    }
+
+    private static DefaultHttpContext CreateRequest()
```

Comment:
Maybe something like `CreateContextWithRequest()`?

Reply:
I'd rather keep it concise for test code.